c_glib/test/test-array.rb

```diff
@@ -131,4 +131,24 @@ def test_invalid
       assert_equal(message, error.message.lines.first.chomp)
     end
   end
+
+  sub_test_case("#diff") do
+    def test_no_diff
+      array = build_string_array(["Start", "Shutdown", "Reboot"])
+      other_array = build_string_array(["Start", "Shutdown", "Reboot"])
+      assert_equal("", array.diff(other_array))
+    end
+
+    def test_diff
+      array = build_string_array(["Start", "Shutdown", "Reboot"])
+      other_array = build_string_array(["Start", "Shutdonw", "Reboot"])
+      assert_equal(<<-STRING.chomp, array.diff(other_array))
+
+@@ -1, +1 @@
+-"Shutdown"
++"Shutdonw"
+
+      STRING
+    end
```

Comment:
Could you also add different type case? For example, string array and int8 array.

Reply:
I've added test cases for int8 array.

Reply:
Ah, I want a test case for different type arrays case. For example, `string_array.diff_unified(int8_array)`.

Reply:
Oh...sorry.  I've also added a test case for different type arrays case.